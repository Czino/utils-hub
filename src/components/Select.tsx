import { FiChevronDown } from 'react-icons/fi'

export const Select = ({ className, children, ...props }: React.ComponentProps<'select'>) => (
  <div className="grid">
    <FiChevronDown className="relative right-1 z-10 col-start-1 row-start-1 justify-self-end self-center w-6 h-6 pointer-events-none" />
    <select
      className={[
        'row-start-1 col-start-1 appearance-none bg-transparent text-dark p-2 border border-dark rounded',
        className,
      ].join(' ')}
      {...props}
    >
      {children}
    </select>
  </div>
)
