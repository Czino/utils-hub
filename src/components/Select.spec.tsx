import { render } from '@testing-library/react'
import { Select } from './Select'

describe('Select', () => {
  it('should render Select with styles', () => {
    const { asFragment } = render(
      <Select className="mt-4" value="text">
        <option>1</option>
        <option>2</option>
      </Select>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
