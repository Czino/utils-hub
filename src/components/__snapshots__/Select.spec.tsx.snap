// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render Select with styles 1`] = `
<DocumentFragment>
  <div
    class="grid"
  >
    <svg
      class="relative right-1 z-10 col-start-1 row-start-1 justify-self-end self-center w-6 h-6 pointer-events-none"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
    <select
      class="row-start-1 col-start-1 appearance-none bg-transparent text-dark p-2 border border-dark rounded mt-4"
    >
      <option>
        1
      </option>
      <option>
        2
      </option>
    </select>
  </div>
</DocumentFragment>
`;
